partner_event/models/event_registration.py

```diff
@@ -35,19 +36,17 @@ def create(self, vals):
             attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
             event = Event.browse(vals["event_id"])
             if attendee_partner:
-                vals["name"] = vals.setdefault("name", attendee_partner.name)
-                vals["phone"] = vals.setdefault("phone", attendee_partner.phone)
+                for key in ["name", "phone", "mobile"]:
+                    if attendee_partner[key]:
```

Comment:
This code is not equivalent, as it's overwriting possible existing values in `vals`, which previous code was not doing.

Reply:
Finally this part has changed because in V12 it did not work (you could not define a value of something that already had a False value by default in the create function).

I have changed the create function because if there is a partner and we associate it, we must force in that record to save the fields of the record (name, phone, mobile).